src/auth/auth.service.ts

```diff
@@ -113,31 +100,26 @@ export class AuthService {
       message: EmailMessages.emailSendSuccess,
     };
   }
+
   async verify(token: string) {
     const payload = this.jwtService.verify<EmailVerificationPayload>(token, {
       secret: process.env.JWT_VERIFICATION_TOKEN_SECRET,
     });
 
-    const record = await this.emailVerificationRepo.findOneByOrFail({
-      token,
+    const record = await this.emailVerificationRepo.findOne({
+      where: { token, expiresAt: LessThan(new Date()) },
     });
 
-    if (record.expiresAt < new Date()) {
-      throw new BadRequestException('Token has expired');
+    if (!record) {
+      throw new NotFoundException('Token not found or has expired');
     }
 
     await this.emailVerificationRepo.save(record);
 
-    const user = await this.userRepository.findOne({
-      where: {
-        email: payload.email,
-      },
-    });
+    const user = await this.userService.findOneByField('email', payload.email);
     if (!user) throw new Error('User not found');
```

Comment:
As I have also been suggested; that the below structure would be better for readability.
```suggestion
    if (!user) 
    {
    throw new Error('User not found');
    }
```

Reply:
updated thank you